UnityProject/Assets/Scripts/Items/Weapons/PowerSink.cs

```diff
@@ -0,0 +1,163 @@
+		private float currentCharge;
+
+		private void Awake()
+		{
+			objectBehaviour = GetComponent<ObjectBehaviour>();
+			spriteHandler = GetComponentInChildren<SpriteHandler>();
+			pickupable = GetComponentInChildren<Pickupable>();
+		}
+
+		private void OnDisable()
+		{
+			UpdateManager.Remove(CallbackType.PERIODIC_UPDATE, CheckForVoltage);
+		}
+
+		public void ServerPerformInteraction(HandApply interaction)
+		{
+			if (interaction.UsedObject == null)
+			{
+				ToggleActivity();
+				return;
+			}
+			if (interaction.UsedObject.Item().HasTrait(CommonTraits.Instance.Screwdriver))
+			{
+				if (isAnchored) UnAnchor();
+				else Anchor();
+				return;
+			}
+			if (interaction.UsedObject.Item().HasTrait(CommonTraits.Instance.Multitool))
+			{
+				Chat.AddExamineMsg(interaction.Performer, $"This device's battery unit is holding {currentCharge.ToEngineering("V")}");
+				return;
+			}
+			if (interaction.UsedObject.TryGetComponent<RemoteSignaller>(out var signaller))
+			{
+				Emitter = signaller;
+				Frequency = signaller.Frequency;
+				Chat.AddExamineMsg(interaction.Performer, $"You pair the {interaction.UsedObject.ExpensiveName()} to this device.");
+			}
+		}
+
+		public bool WillInteract(HandApply interaction, NetworkSide side)
+		{
+			if (DefaultWillInteract.Default(interaction, side) == false) return false;
+			return true;
+		}
+
+		private void Anchor()
+		{
+			isAnchored = true;
+			pickupable.ServerSetCanPickup(false);
+			objectBehaviour.ServerSetPushable(false);
+			Chat.AddLocalMsgToChat($"The {gameObject.ExpensiveName()} makes a clicking sound as it anchors to the ground", gameObject);
+			SoundManager.PlayNetworkedAtPos(CommonSounds.Instance.screwdriver, gameObject.AssumedWorldPosServer());
+		}
+		private void UnAnchor()
+		{
+			isAnchored = false;
+			pickupable.ServerSetCanPickup(true);
+			objectBehaviour.ServerSetPushable(true);
+			UpdateManager.Remove(CallbackType.PERIODIC_UPDATE, CheckForVoltage);
+			Chat.AddLocalMsgToChat($"The {gameObject.ExpensiveName()} makes a clicking sound as it unanchors from the ground", gameObject);
+			SoundManager.PlayNetworkedAtPos(CommonSounds.Instance.screwdriver, gameObject.AssumedWorldPosServer());
+		}
+
+		private void ToggleActivity()
+		{
+			if (isAnchored == false) return;
+			if (isCharging)
+			{
+				UpdateManager.Remove(CallbackType.PERIODIC_UPDATE, CheckForVoltage);
+				spriteHandler.SetSpriteSO(inactiveSpriteSO);
+			}
+			else
+			{
+				UpdateManager.Add(CheckForVoltage, voltageCheckTimeInSeconds);
+				spriteHandler.SetSpriteSO(activeSpriteSO);
+			}
+			isCharging = !isCharging;
+		}
+
+		private void Detonate()
+		{
+			// Get data before despawning
+			var worldPos = gameObject.AssumedWorldPosServer();
+			// Despawn the explosive
+			_ = Despawn.ServerSingle(gameObject);
+			Explosion.StartExplosion(worldPos.RoundToInt(), currentCharge * explosionAmplifer);
+		}
+
+		public void CheckForVoltage()
+		{
+			if (isAnchored == false) return;
```

Comment:
does this mean it doesn't actually draw power?

Reply:
i've given up on trying to do that plus it just complicated the explosion calc so right now this is good as it is